src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -329,6 +368,28 @@ module.exports = class LevelPlane extends Array {
   }
 
   /**
+  * Find all iron doors in a level and evaluate if they need to be animated based on state
+  */
+  findDoorToAnimate(notOffendingIndex) {
+    for (let i = 0; i < this.length; ++i) {
```

Comment:
Let's iterate through the positions so we're not directly dealing with indices until we need to actually look up items in the array. See my feedback here: https://github.com/code-dot-org/craft/pull/67/files#r137057705.

Reply:
Do you mean like, 
for(i = 0....)
 for(j = 0...)
  if(this.getblockat([i,j]).blockType .... )

Or is there some better way to go through all the positions in the plane? I just did it that way because at the end of the day, the array is 1D so it seemed an easy way to just cover every index.